### PR TITLE
Update interface to allow different learning rate schedulers

### PR DESCRIPTION
**Pull request by mo-sameh**

### CODE REVIEW

File: peptdeep/model/model_interface.py [hunk 1]
```diff
@@ -32,56 +32,125 @@
...
-    optimizer, num_warmup_steps: int, num_training_steps: int, num_cycles: float = 0.5, last_epoch: int = -1
-):
+class LR_SchedulerInterface(object):
+    def __init__(self, optimizer:torch.optim.Optimizer, **kwargs):
+        raise NotImplementedError
+
+    def step(self, epoch:int, loss:float):
+        """
+        This method must be implemented in the sub-class. It will be called to get the learning rate for the next epoch. 
+        While the one we are using here does not need the loss value, this is left incase of using something like the ReduceLROnPlateau scheduler.
```
Comment (GeorgWa) [inline peptdeep/model/model_interface.py:42]: (Nit) typo in case

File: peptdeep/model/model_interface.py [hunk 2]
```diff
@@ -185,6 +255,24 @@ def target_column_to_train(self)->str:
     def target_column_to_train(self, column:str):
         self._target_column_to_train = column
 
+    def set_lr_scheduler_class(self, lr_scheduler_class:LR_SchedulerInterface) -> None:
+        """
+        Set the learning rate scheduler class. We require the user pass a class that is a subclass of 
+        LR_SchedulerInterface because the current implementation will create an instance of it within this class.
```
Comment (GeorgWa) [inline peptdeep/model/model_interface.py:261]: Just to be sure I understand it correctly. We provide a Class instead of an object because we might not have access to the optimizer when creating the object outside?

**Review by GeorgWa — APPROVED** [review]
LGTM I think this addresses all comments from #154

File: peptdeep/model/model_interface.py (hunk 2 above)
Comment (jalew188) [reply peptdeep/model/model_interface.py:261]: I think it is better to define and instansialize `self.lr_scheduler` in the `train()` function. 

I guess the purpose of using `lr_scheduler_class` is to leave an interface for other lr_schedulers in the future; however, lr_scheduler's `init` method requires the corresponding init args from `train()` method. If lr_scheduler is changed, users need to change train() args at the mean time, meaning that these two functionalities are not decoupled well.

Comment (jalew188) [reply peptdeep/model/model_interface.py:261]: Now I understand, but I think this leaves a potential bug. We can create a new ModelInterface .py file and then we can slowly copy/move functionalities there.

**Review by jalew188 — APPROVED** [review]
